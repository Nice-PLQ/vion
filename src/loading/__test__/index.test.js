import { mount } from '@vue/test-utils';
import Loading from '..';

describe('Loading', () => {
  test('普通的loading', () => {
    const wrapper = mount({
      template: '<v-loading>正在加载..</v-loading>'
    }, {
      global: {
        components: { VLoading: Loading }
      }
    });
    const loading = wrapper.find('.v-loading');
    expect(loading.element).toMatchSnapshot();
    expect(loading.classes()).toContain('v-loading_normal');
    expect(loading.find('.v-loading__txt-tips').text()).toBe('正在加载..');
  });

  test('悬浮的loading', async () => {
    const wrapper = mount({
      template: '<v-loading />'
    }, {
      props: {
        type: 'float',
        text: '正在加载',
        visible: false,
        duration: 1000
      },
      global: {
        components: { VLoading: Loading }
      }
    });
    await wrapper.setProps({ visible: true });
    const loading = wrapper.find('.v-loading');
    expect(loading.element).toMatchSnapshot();
    expect(loading.classes()).toContain('v-loading_float');
    expect(loading.find('.v-loading__txt-tips').text()).toBe('正在加载');
  });
});
