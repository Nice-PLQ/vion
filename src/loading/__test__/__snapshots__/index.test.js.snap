// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading 悬浮的loading 1`] = `
<div
  class="v-loading v-loading_float"
>
  <div
    class="v-loading__inner"
  >
    <div
      class="v-loading__icon"
    />
    <span
      class="v-loading__txt-tips"
    >
      正在加载
    </span>
  </div>
</div>
`;

exports[`Loading 普通的loading 1`] = `
<div
  class="v-loading v-loading_normal"
>
  <div
    class="v-loading__inner"
  >
    <div
      class="v-loading__icon"
    />
    <span
      class="v-loading__txt-tips"
    >
      
      正在加载..
      
    </span>
  </div>
</div>
`;
